src/test/matching_test.cpp

```diff
@@ -40,24 +40,28 @@ class TestAutoIndex : public MinIndexSelection {
 
 using Nodes = set<SearchSignature>;
 
+void setBits(SearchSignature& search, uint64_t mask) {
+    size_t len = search.arity();
+    for (size_t i = 0; i < len; ++i) {
+        if (mask % 2) {
+            search.set(i, AttributeConstraint::Equal);
+        }
+        mask /= 2;
+    }
+}
+
 TEST(Matching, StaticTest_1) {
     TestAutoIndex order;
     Nodes nodes;
     size_t arity = 5;
-    order.addSearch(SearchSignature(arity, 31));
-    nodes.insert(SearchSignature(arity, 31));
-    order.addSearch(SearchSignature(arity, 23));
-    nodes.insert(SearchSignature(arity, 23));
-    order.addSearch(SearchSignature(arity, 15));
-    nodes.insert(SearchSignature(arity, 15));
-    order.addSearch(SearchSignature(arity, 7));
-    nodes.insert(SearchSignature(arity, 7));
-    order.addSearch(SearchSignature(arity, 5));
-    nodes.insert(SearchSignature(arity, 5));
-    order.addSearch(SearchSignature(arity, 3));
-    nodes.insert(SearchSignature(arity, 3));
-    order.addSearch(SearchSignature(arity, 1));
-    nodes.insert(SearchSignature(arity, 1));
+
+    std::array<uint64_t, 7> patterns = {1, 3, 5, 7, 15, 23, 31};
+    for (auto pattern : patterns) {
+        SearchSignature search(arity);
+        setBits(search, pattern);
```

Comment:
why not just have `setBits` helper return the signature? (`auto src = setBits(arity, pat)`)

Reply:
Fixed